test/e2e/kubectl.go

```diff
@@ -529,20 +529,28 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 
 	framework.KubeDescribe("Kubectl should be able to talk to api server", func() {
 		It("kubectl runing in a pod should talk to api server [Conformance]", func() {
+			framework.SkipUnlessProviderIs("gke")
 			nsFlag := fmt.Sprintf("--namespace=%v", ns)
 			podJson := readTestFileOrDie(kubectlInPodFilename)
 			By("validating api verions")
 			framework.RunKubectlOrDieInput(string(podJson), "create", "-f", "-", nsFlag)
-			for i := 0; i < 20; i++ {
+			err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
 				output := framework.RunKubectlOrDie("get", "pods/kubectl-in-pod", nsFlag)
-				if !strings.Contains(output, "ContainerCreating") {
-					break
+				if strings.Contains(output, "Running") {
+					return true, nil
+				} else {
+					return false, nil
 				}
+			})
+			if err != nil {
```

Comment:
It seems [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl.go#L503) has the same problem.
Can I update it here or open another PR for it?